docs/custom-filters.adoc

```diff
@@ -453,11 +453,13 @@ public class FetchFilter implements FetchRequestFilter {
 <1> We construct a new request object.  This will be forwarded asynchronously.
 <2> We send the request towards the Cluster, specifying the api version to use and obtaining a CompletionStage for the
 response.
-<3> We use CompletionStage chaining to mutate the filtered fetch request using the response from the request sent at <1>.
-<4> We use CompletionStage chaining to forward the mutated request.
+<3> We use a computation stage to mutate the filtered fetch request using the response from the request sent at <1>.
+<4> We use another computation stage to forward the mutated request.
 
-NOTE: Kroxylicious provides the guarantee that the chained stage is completed on the same thread as the rest of the
-Filter work, so we can safely mutate Filter members without synchronising.
+NOTE: Kroxylicious provides the guarantee that computation stages are executed on the same thread as the rest of the
```

Comment:
I read the javadoc in `FilterContext` as making a weaker (but more accurate) guarantee that we only ensure  single thread when no external executor is supplied. 

Should we soften the language here or is that kinda implicit if you supply your own executor thread safety becomes your problem?

Reply:
Intent: I believe we want to avoid duplication in the documentation between asciidoc/javadoc.  So I think we want to use the asciidoc documentation to describe the broad picture and use links to the Javadoc to describe the nitty gritty.  Does this seem reasonable? I was trying to walk that line with this change.

Reply:
Softening done.  I used the phrase "chained using the default execution methods"  as a catch all for methods without the suffix async and methods with suffix async that employ the  default asynchronous execution facility.. IOW all the chaining methods that use custom asynchronous execution.  The reference to the Javadoc should let the reader quickly understand exactly what is meant.